runPerftool.psm1

```diff
@@ -339,36 +380,52 @@ param(
+            $sw.Reset()
+            $sw.Start()
+            while (([math]::Round($sw.Elapsed.TotalSeconds,0)) -lt $timeout){
+
+                start-sleep -seconds 5
+
                 $checkRecvProcessExit = Invoke-Command -Session $recvPSSession -ScriptBlock $CheckProcessExitScriptBlock -ArgumentList "$Toolname"
                 $checkSendProcessExit = Invoke-Command -Session $sendPSSession -ScriptBlock $CheckProcessExitScriptBlock -ArgumentList "$Toolname"
 
                 if (($checkRecvProcessExit -eq $null)-and ($checkSendProcessExit -eq $null)){
+                    $cmdPairCompleted = $true
                     LogWrite "$Toolname exited on both Src and Dest machines"
                     break
                 }
-                start-sleep -seconds 5
             }
 
-        $checkRecvProcessExit = Invoke-Command -Session $recvPSSession -ScriptBlock $CheckProcessExitScriptBlock -ArgumentList "$Toolname"
-        $checkSendProcessExit = Invoke-Command -Session $sendPSSession -ScriptBlock $CheckProcessExitScriptBlock -ArgumentList "$Toolname"
+            $sw.Stop()
+            
+            #Wait for disk I/O to be completed
+            Write-VolumeCache (get-location).Drive.Name
+            start-sleep -seconds 30
 
-        if ($checkRecvProcessExit -ne $null) {
-            LogWrite " ++ $Toolname on Receiver did not exit cleanly... "
-        }
-        if ($checkSendProcessExit -ne $null) {
-            LogWrite " ++ $Toolname on Sender did not exit cleanly... "
-        }
+            # If command pair didnt gracefully exit, do the logging, cleanup here
+            if(-Not $cmdPairCompleted) {
+                $checkRecvProcessExit = Invoke-Command -Session $recvPSSession -ScriptBlock $CheckProcessExitScriptBlock -ArgumentList "$Toolname"
+                $checkSendProcessExit = Invoke-Command -Session $sendPSSession -ScriptBlock $CheckProcessExitScriptBlock -ArgumentList "$Toolname"
 
-        #Since time is up, clean up any processes that failed to exit gracefully so that the new commands can be issued
-        Invoke-Command -Session $recvPSSession -ScriptBlock $ScriptBlockTaskKill -ArgumentList $toolexe
-        Invoke-Command -Session $sendPSSession -ScriptBlock $ScriptBlockTaskKill -ArgumentList $toolexe
+                if ($checkRecvProcessExit -ne $null) {
+                    LogWrite (" ++ {0} on Receiver did not exit cleanly... Timer Elapsed Value: {1}" -f $Toolname, ($sw.elapsed.TotalSeconds))
+                }
+                if ($checkSendProcessExit -ne $null) {
+                    LogWrite (" ++ {0} on Sender did not exit cleanly...  Timer Elapsed Value: {1}" -f $Toolname, ($sw.elapsed.TotalSeconds))
+                }
+            }
+
+            #Since time is up, clean up any processes that failed to exit gracefully so that the new commands can be issued
+            Invoke-Command -Session $recvPSSession -ScriptBlock $ScriptBlockTaskKill -ArgumentList $toolexe
```

Comment:
Visually looks like a dup line.  Why not just loop and iterate on the $sendPSSession and $recvPSSession?